libyul/backends/wasm/WasmDialect.cpp

```diff
@@ -85,6 +87,69 @@ WasmDialect const& WasmDialect::instance()
 	return *dialect;
 }
 
+void WasmDialect::addEthereumExternals()
+{
+	// These are not YulStrings because that would be too complicated with regards
+	// to the YulStringRepository reset.
+	static string const i64{"i64"};
+	static string const i32{"i32"};
+	static string const i32ptr{"i32"}; // Uses "i32" on purpose.
+	struct External { string name; vector<string> parameters; vector<string> returns; };
+	static vector<External> externals{
+		{"getAddress", {i32ptr}, {}},
+		{"getExternalBalance", {i32ptr, i32ptr}, {}},
+		{"getBlockHash", {i64, i32ptr}, {i32}},
```

Comment:
what's the output here?

Reply:
`getBlockhash(blockNr, outputPtr) -> success`

Reply:
Acutally output is 1 on failure and 0 on success.